service/src/main/java/skills/auth/saml/SAML2SecurityConfiguration.groovy

```diff
@@ -0,0 +1,178 @@
+import skills.auth.PortalWebSecurityHelper
+import skills.auth.SecurityMode
+import skills.auth.UserInfo
+import skills.auth.form.LocalUserDetailsService
+
+import java.security.KeyFactory
+import java.security.PrivateKey
+import java.security.cert.CertificateFactory
+import java.security.cert.X509Certificate
+import java.security.spec.PKCS8EncodedKeySpec
+
+
+@Slf4j
+@Conditional(SecurityMode.SAML2Auth)
+@Component
+@Configuration
+@EnableWebSecurity
+/** Supports Single Identity Provider Only **/
+class SAML2SecurityConfiguration{
+
+    public static final String SKILLS_REDIRECT_URI = 'skillsRedirectUri'
+
+    @Value('${spring.security.saml2.metadata-location}')
+    String assertingPartyMetadataLocation;
+
+    @Value('${spring.security.saml2.registrationId}')
+    String registrationId;
+
+    // Custom properties using for configuring signing credentials
+
+    @Value('${saml2.rp.signing.key-location:#{null}}')
+    private String privateKeyLocation;
+
+    @Value('${saml2.rp.signing.cert-location:#{null}}')
+    private String certificateLocation;
+
+    @Autowired
+    PortalWebSecurityHelper portalWebSecurityHelper
+
+    @Autowired
+    SAML2Utils saml2Utils;
+
+    private final ResourceLoader resourceLoader;
+
+    @Autowired
+    public SAML2SecurityConfiguration(ResourceLoader resourceLoader) {
+        this.resourceLoader = resourceLoader;
+    }
+
+
+     @Bean
+    public SecurityFilterChain filterChain(HttpSecurity http) throws Exception   {
+
+         OpenSaml4AuthenticationProvider authenticationProvider = new OpenSaml4AuthenticationProvider();
+         authenticationProvider.setResponseAuthenticationConverter(responseToken -> {
+             Saml2Authentication authentication = OpenSaml4AuthenticationProvider.createDefaultResponseAuthenticationConverter()
+                                                                                 .convert(responseToken);
+             Saml2AuthenticatedPrincipal principal = (Saml2AuthenticatedPrincipal) authentication.getPrincipal();
+            // Assertion assertion = responseToken.getResponse().getAssertions().get(0);
+             UserInfo userInfo = saml2Utils.convertToUserInfo(authentication,true);
+
+             return new Saml2Authentication(principal,authentication.getSaml2Response(),userInfo.getAuthorities());
+         });
+
+         portalWebSecurityHelper.configureHttpSecurity(http);
+         http.saml2Login(saml2 -> saml2
+                 .authenticationManager(new ProviderManager(authenticationProvider))
+                 .relyingPartyRegistrationRepository(relyingPartyRegistrationRepository())
+         );
+         http.logout((logout) -> logout.logoutSuccessUrl("/local/logout"));
+         return http.build();
+    }
+   @Bean
+    RelyingPartyRegistrationRepository relyingPartyRegistrationRepository() {
+        RelyingPartyRegistration registration = null;
+
+        if(privateKeyLocation!=null && certificateLocation!=null){
```

Comment:
It appears that `loadPrivateKey` and `loadCertificate` load a key and a cert from classpath and `loadPrivateKey` replaces contents of the actual key; there should not be production code loading key/cert from the classpath; looks like this code was added as a workaround to get `SAML2IntegrationIT` passing and is not needed for the production runtime; please remove `if(privateKeyLocation!=null && certificateLocation!=null){` statement and its scope as well as auto wiring of these properties; `SAML2IntegrationIT` should be configured to utilize main runtime code of 
```
registration = RelyingPartyRegistrations.fromMetadataLocation(assertingPartyMetadataLocation)
                    .registrationId(registrationId)
                    .build()
```

Reply:
loadPrivateKey and loadCertificate methods retrieve key and cert from both classpath and file system.  Added these to support use cases wherein signing of auth requests is mandatory.  Refactored code to check the value of attribute named 'WantAuthnRequestsSigned' and conditionally load the key and cert for signing the saml request